### PR TITLE
Use BitBagCmsPlugin in Shop

### PR DESCRIPTION
**Pull request by Tomanhez**
Before:
Screenshot 2020-01-15 at 10 55 37
After:
Screenshot 2020-01-15 at 10 57 10

Before: 
Screenshot 2020-01-15 at 11 01 27
Screenshot 2020-01-15 at 11 01 57
Screenshot 2020-01-15 at 11 02 45
Screenshot 2020-01-15 at 10 57 19
Screenshot 2020-01-15 at 11 01 06
Screenshot 2020-01-15 at 11 07 11

### CODE REVIEW

File: templates/bundles/SyliusShopBundle/Homepage/index.html.twig [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{% extends '@SyliusShop/layout.html.twig' %}
+
+{% block content %}
+    <div class="homepage">
+        <div class="ui hidden divider"></div>
+            {{ bitbag_cms_render_media('homepage_banner') }}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/index.html.twig]: Should be moved to https://github.com/Sylius/Sylius/pull/11014/files#diff-cd30ca237d9d140b0ba8594e7645c29eR90

File: templates/bundles/SyliusShopBundle/Homepage/index.html.twig [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+        <h2 class="ui center aligned huge header">{{ 'sylius.ui.latest_products'|trans }}</h2>
+        <div class="ui hidden divider"></div>
+
+        <div {{ sylius_test_html_attribute('latest-products') }}>
+            {{ render(url('sylius_shop_partial_product_index_latest', {'count': 3, 'template': '@SyliusShop/Homepage/_list.html.twig'})) }}
+        </div>
+        <div class="ui hidden divider"></div>
+
+        <h2 class="ui center aligned huge header">{{ 'sylius.homepage.brace_yoursel_winter_is_coming'|trans }}</h2>
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/index.html.twig]: This line (and 3 lines before) should be moved to https://github.com/Sylius/Sylius/pull/11014/files#diff-cd30ca237d9d140b0ba8594e7645c29eR96

File: templates/bundles/SyliusShopBundle/Homepage/index.html.twig [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        <div class="ui hidden divider"></div>
+
+        {% include '@SyliusShop/Homepage/_carousel.html.twig' %}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/index.html.twig]: I think we should be able to just customize the way block is rendered in this case 

ref. https://github.com/BitBagCommerce/SyliusCmsPlugin/blob/master/doc/blocks.md#general-usage

File: templates/bundles/SyliusShopBundle/Homepage/index.html.twig [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        <div class="ui hidden divider"></div>
+
+        <div class="ui very padded center aligned secondary segment">
+            <div class="ui centered grid">
+                <div class="sixteen wide tablet twelve wide computer center aligned column">
+                    <h2 class="ui huge header">
+                        {{ 'sylius.homepage.about_us'|trans }}
+                    </h2>
+                    <p>
+                        {{ bitbag_cms_render_block('about_us') }}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/index.html.twig]: Should be moved to https://github.com/Sylius/Sylius/pull/11014/files#diff-cd30ca237d9d140b0ba8594e7645c29eR105

File: templates/bundles/SyliusShopBundle/Homepage/index.html.twig [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                {{ render(url('sylius_shop_partial_product_index_latest', {'count': 4, 'template': '@SyliusShop/Homepage/_list.html.twig'})) }}
+            </div>
+        </div>
+        <div class="ui hidden divider"></div>
+
+        <div class="ui very padded center aligned secondary segment">
+            <div class="ui centered grid">
+                <div class="sixteen wide tablet twelve wide computer center aligned column">
+                    <h2 class="ui huge header">
+                        {{ 'sylius.homepage.about_us'|trans }}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/index.html.twig]: Should be part of the `about_us` block

File: templates/bundles/SyliusShopBundle/Product/_box.html.twig [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{% import "@SyliusShop/Common/Macro/money.html.twig" as money %}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Product/_box.html.twig]: Is this file needed?

File: src/Twig/ReturnCmsBlockExtension.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+declare(strict_types=1);
+
+namespace App\Twig;
+
+use BitBag\SyliusCmsPlugin\Entity\BlockInterface;
+use BitBag\SyliusCmsPlugin\Resolver\BlockResourceResolverInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class ReturnCmsBlockExtension extends AbstractExtension
```
Comment (lchrusciel) [inline src/Twig/ReturnCmsBlockExtension.php]: If we will manage to use the custom block template, then this extension will not be needed

File: templates/bundles/SyliusShopBundle/Homepage/_carousel.html.twig [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<div class="carousel-wrapper">
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/_carousel.html.twig]: If we will manage to use the custom block template, then this file will not be needed

File: templates/bundles/SyliusShopBundle/Homepage/index.html.twig [hunk 9]
```diff
@@ -0,0 +1,72 @@
+{% extends '@SyliusShop/layout.html.twig' %}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/index.html.twig]: Please, update to the newest master and use template events instead of overriding of this template ;)

File: templates/bundles/SyliusShopBundle/Homepage/Block/_latestProductsCarousel.html.twig [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<h2 class="ui center aligned huge header">{{ 'sylius.homepage.brace_yoursel_winter_is_coming'|trans }}</h2>
+<div class="ui hidden divider"></div>
+
+{{ render(path('bitbag_sylius_cms_plugin_shop_block_render', {'code' : 'carousel', 'template' : '@SyliusShop/Homepage/_carousel.html.twig'})) }}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/Block/_latestProductsCarousel.html.twig]:
```suggestion
{{ render(path('bitbag_sylius_cms_plugin_shop_block_render', {'code' : 'carousel', 'template' : 'Shop/Homepage/Block/_carousel.html.twig'})) }}
```

File: templates/bundles/SyliusShopBundle/Homepage/Block/_latestProductsCarousel.html.twig [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<h2 class="ui center aligned huge header">{{ 'sylius.homepage.brace_yoursel_winter_is_coming'|trans }}</h2>
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/Block/_latestProductsCarousel.html.twig]: Let's move these lines to `_carousel.html.twig`

File: templates/bundles/SyliusShopBundle/Homepage/Block/_latestProductsCarousel.html.twig [hunk 12]
```diff
@@ -0,0 +1,4 @@
...
+<div class="ui hidden divider"></div>
+
+{{ render(path('bitbag_sylius_cms_plugin_shop_block_render', {'code' : 'seasonal_promotion', 'template' : '@SyliusShop/Homepage/_carousel.html.twig'})) }}
```
Comment (lchrusciel) [inline templates/bundles/SyliusShopBundle/Homepage/Block/_latestProductsCarousel.html.twig]: IMHO 
```suggestion
{{ render(path('bitbag_sylius_cms_plugin_shop_block_render', {'code' : 'seasonal_promotion', 'template' : 'Shop/Homepage/Blocks/_carousel.html.twig'})) }}
```

and the file itself should be placed in `templates/Shop/Homepage/Blocks/_carousel.html.twig`

File: templates/bundles/SyliusShopBundle/Homepage/_carousel.html.twig (hunk 8 above)
Comment (Tomanhez) [reply templates/bundles/SyliusShopBundle/Homepage/_carousel.html.twig]: this file is require at this moment

**Comment by GSadee** [comment]
And rebase it with master

File: templates/Shop/Homepage/Block/_carousel.html.twig [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<h2 class="ui center aligned huge header">{{ 'sylius.homepage.brace_yoursel_winter_is_coming'|trans }}</h2>
```
Comment (lchrusciel) [inline templates/Shop/Homepage/Block/_carousel.html.twig]: I'm afraid we should take a text from block here  according to our DoD definition

**Comment by Zales0123** [comment]
Thank you, Tomasz! :1st_place_medal:
